highlighters/main/main-highlighter.zsh

```diff
@@ -650,11 +652,26 @@ _zsh_highlight_main_highlighter_highlight_list()
    fi
 
    # The Great Fork: is this a command word?  Is this a non-command word?
-   if [[ -n ${(M)ZSH_HIGHLIGHT_TOKENS_COMMANDSEPARATOR:#"$arg"} ]]; then
+   if [[ $this_word == :test: ]]; then
+     next_word=:test:
+     if [[ $arg == $'\x5d\x5d' ]]; then
+       _zsh_highlight_main__stack_pop T reserved-word
+       next_word=:end:
+     elif [[ $arg == ('&&'|'||') ]]; then
+       style=reserved-word
+     elif [[ -n ${(M)ZSH_HIGHLIGHT_TOKENS_COMMANDSEPARATOR:#"$arg"} ]]; then
+       style=unknown-token
+       next_word=:start:
+       highlight_glob=true
+     else
```

Comment:
Why `1` (the first)?

The second `1` may want to be converted to `true`/`false` or Perl-style named varargs for readability.